File: api/internal/domain/models/game.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Game struct {
	ID        string    `json:"id"`
	LeagueID  string    `json:"leagueId"`
	StartTime time.Time `json:"startTime"`
	Location  string    `json:"location"`

	CostInCents int  `json:"costInCents"`
	IsCanceled  bool `json:"isCanceled"`

	Attendance []GameAttendance `json:"attendance"`
	Charges    []GameCharge     `json:"charges"`
}

func CreateGame(leagueId, location string, costInCents int) *Game {
	gameId, _ := uuid.NewV7()

	return &Game{
		ID:          gameId.String(),
		LeagueID:    leagueId,
		StartTime:   time.Now(),
		Location:    location,
		CostInCents: costInCents,
		IsCanceled:  false,
		Attendance:  make([]GameAttendance, 0),
		Charges:     make([]GameCharge, 0),
	}
}

File: bouncy-web/src/lib/models/game.ts
import type { GameAttendance } from './game-attendance';
import type { GameCharge } from './game-charge';

export interface Game {
  id: string;
  leagueId: string;
  startTime: Date;
  location: string;
  costInCents: number;
  isCanceled: boolean;
  attendance: GameAttendance[];
  charges: GameCharge[];
}
